app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -1150,14 +1152,46 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
 
   @OnClick(R.id.tab_switcher_close_all_tabs)
   void closeAllTabs() {
+    rotate(closeAllTabsButton);
     webViewList.clear();
     tabsAdapter.notifyDataSetChanged();
+    openHomeScreen();
+    updateTabSwitcherIcon();
+  }
+
+  //opens home screen when user closes all tabs
+  private void openHomeScreen() {
     new Handler().postDelayed(() -> {
       if (webViewList.size() == 0) {
         newTab(HOME_URL);
+        hideTabSwitcher();
       }
-    }, 1500);
-    updateTabSwitcherIcon();
+    }, 100);
+  }
+
+  public void rotate(FloatingActionButton v) {
```

Comment:
Can you move this method to appropriate class in utils? This class is already 2k lines long. You will require `Context` which you can get from the view itself.

Reply:
@abdulwd I have made the changes. Please review it once.